cljfmt/src/cljfmt/core.cljc

```diff
@@ -118,18 +120,24 @@
    :set "#{", :deref "@",  :reader-macro "#", :unquote "~"
    :var "#'", :quote "'",  :syntax-quote "`", :unquote-splicing "~@"})
 
-(defn- prior-string [zloc]
-  (if-let [p (zip/left zloc)]
-    (str (prior-string p) (n/string (z/node p)))
-    (if-let [p (zip/up zloc)]
-      (str (prior-string p) (start-element (n/tag (z/node p))))
-      "")))
+(defn- prior-line-string [zloc]
+  (loop [zloc     zloc
+         worklist '()]
+    (if-let [p (zip/left zloc)]
+      (let [s (str (n/string (z/node p)))]
+        (if-not (str/includes? s "\n")
```

Comment:
Yes but we need another reader conditional. Which is doable, just not the first thing I'd reach for.